src/Symfony/Component/DependencyInjection/Compiler/AutowirePass.php

```diff
@@ -273,7 +273,7 @@ private function autowireMethod(\ReflectionFunctionAbstract $reflectionMethod, a
                         break;
                     }
 
-                    if (InnerService::class === $attribute->getName()) {
+                    if (Decorated::class === $attribute->getName()) {
```

Comment:
`if (Decorated::class === $attribute->getName() && $decorated = $this->container->getDefinition($this->currentId)->getDecoratedService()) {` I suppose

Reply:
adding the second condition breaks the tests added for this feature, not sure why as it's been a long time I didn't dig into this area. Let me know if it's ok or if I need to investigate

Reply:
got it https://github.com/symfony/symfony/blob/6.1/src/Symfony/Component/DependencyInjection/Compiler/DecoratorServicePass.php#L48
the decoration info will never be available at this stage

Reply:
change reverted

Reply:
ok just found what we're looking for:
https://github.com/symfony/symfony/blob/84d35a2c9adb51b84c81889e351006f89f306049/src/Symfony/Component/DependencyInjection/Compiler/DecoratorServicePass.php#L57-L58
should be all good now